File: Chapter7Search/DSC++/Edge.cpp
#pragma once
#include "Node.cpp"
class Edge {
private:
	long value;
	Node leftLink;
public:
	Edge(long value,Node leftLink) {
		this->leftLink = leftLink;
		this->value = value;
	}

	long getValue() {
		return this->value;
	}

	Node getLeftLink() {
		return leftLink;
	}

	bool operator==(const Edge& e) const {
		return (value == e.value) && (leftLink == e.leftLink);
	}

	~Edge() {
		
	}
};

File: Chapter7Search/DSJava/Edge.java
package com.company;

public class Edge {
    private long value;
    private Node leftLink = null;

    public Edge(long value, Node leftLink) {
        this.value = value;
        this.leftLink = leftLink;
    }

    public void setLeftLink(Node leftLink) {
        this.leftLink = leftLink;
    }

    public long getValue() {
        return value;
    }

    public Node getLeftLink() {
        return leftLink;
    }
}
